File: src/frontend/src/types/god.ts
export interface Alias {
  id: number;
  godId: number;
  name: string;
}

export interface God {
  id: number;
  name: string;
  description: string;
  mythologyId: number;
  aliases: Alias[];
}

export interface GodInput {
  id?: number;
  name: string;
  description: string;
  mythologyId: number;
}

File: src/Gods/Models/God.cs
using System.ComponentModel.DataAnnotations;

namespace MythApi.Gods.Models;

public class GodInput {
    public int? Id { get; set; }

    [Required]
    [MaxLength(200)]
    public string Name { get; set; } = null!;

    [Required]
    [MaxLength(2000)]
    public string Description { get; set; } = null!;

    public int MythologyId { get; set; }
}
